crates/pathfinder/src/rpc/v02/method/syncing.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{
+    core::{StarknetBlockHash, StarknetBlockNumber},
+    rpc::{serde::StarknetBlockNumberAsHexStr, v02::RpcContext},
+};
+
+use serde::Serialize;
+
+crate::rpc::error::generate_rpc_error_subset!(SyncingError);
+
+#[allow(dead_code)]
+pub async fn syncing(context: std::sync::Arc<RpcContext>) -> Result<SyncingOuput, SyncingError> {
+    // Scoped so I don't have to think too hard about mutex guard drop semantics.
+    let value = { context.sync_status.status.read().await.clone() };
+
+    use crate::rpc::v01::types::reply::Syncing;
+    let value = match value {
+        Syncing::False(_) => SyncingOuput::False,
+        Syncing::Status(status) => {
+            let status = SyncingStatus {
+                starting_block_num: status.starting.number,
+                current_block_num: status.current.number,
+                highest_block_num: status.highest.number,
+                starting_block_hash: status.starting.hash,
+                current_block_hash: status.current.hash,
+                highest_block_hash: status.highest.hash,
+            };
+            SyncingOuput::Status(status)
+        }
+    };
+
+    Ok(value)
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(any(test, feature = "rpc-full-serde"), derive(serde::Deserialize))]
```

Comment:
Do we still need this? Probably not?

Reply:
We only need this if we want to use this type in end-to-end tests or the load test. I'd think we'll need this in the load test?

Reply:
Ah right, thanks.

Okay I'll leave it be for now. We should probably do an audit of our features instead and remove these that way if required.